plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -518,6 +531,54 @@ def _extract_rpm_data(type_id, rpm_filename):
     return rpm_data
 
 
+def _extract_drpm_data(drpm_filename):
+    """
+    Extract a dict of information for a given DRPM.
+
+    :param drpm_filename: full path to the package to analyze
+    :type  drpm_filename: str
+
+    :return: dict of data about the package
+    :rtype:  dict
+    """
+    drpm_data = dict()
+
+    headers = rpm_parse.package_headers_drpm(drpm_filename)
+
+    try:  # "handle" rpm-only drpms (without rpm header)
+        rpm_headers = rpm_parse.package_headers(drpm_filename)
+    except rpm.error:
+        raise RPMOnlyDRPMsAreNotSupported(drpm_filename)
+
+    drpm_data['signature'] = rpm_parse.package_signature(rpm_headers)
+    drpm_data['arch'] = rpm_headers['arch']
+
+    old_nevr = old_name, old_epoch, old_version, old_release = rpm_parse.nevr(headers["old_nevr"])
+    new_nevr = new_name, new_epoch, new_version, new_release = rpm_parse.nevr(headers["nevr"])
+
+    drpm_data['sequence'] = headers["old_nevr"] + "-" + headers["seq"]
+
+    drpm_data['epoch'] = str(new_epoch)
+    drpm_data['oldepoch'] = str(old_epoch)
+
+    drpm_data['version'] = str(new_version)
+    drpm_data['oldversion'] = str(old_version)
+
+    drpm_data['release'] = new_release
+    drpm_data['oldrelease'] = old_release
+
+    drpm_data['new_package'] = new_name
+    drpm_data['size'] = os.stat(drpm_filename)[stat.ST_SIZE]
+
+    old_evr = rpm_parse.nevr_to_evr(*old_nevr)
+    new_evr = rpm_parse.nevr_to_evr(*new_nevr)
+    drpm_data['filename'] = "drpms/%s-%s_%s.%s.drpm" % (new_name, rpm_parse.evr_to_str(*old_evr),
```

Comment:
@mhrivnak do we want to extract the filename details from the package metadata or we want to keep the real filename?
for example when i create a DRPM
`makedeltarpm -v  ~/Downloads/yum-3.4.3-9.fc16.noarch.rpm ~/Downloads/yum-3.4.3-10.fc16.noarch.rpm yum-test.drpm`
if i sync it then the filename would be retrieved from prestodelta from `filename` tag which would contain `drpms/yum-test.drpm`
if i upload that drpm into the filename would be saved as you can see from Line#575
As a result 2 units will be saved into db with 2 storage paths because `filename` is part of the unit key.
which approach we want to take? do we want to reconstruct the filename from the metadata or use the real one?